cpp/src/cylon/compute/aggregates.cpp

```diff
@@ -253,6 +252,192 @@ cylon::Status Max(const std::shared_ptr<cylon::Table> &table, int32_t col_idx, s
+    }
+    case arrow::Type::UINT64: {
+      return &ResolveTableFromScalar<arrow::UInt64Type>;
+    }
+    case arrow::Type::INT64: {
+      return &ResolveTableFromScalar<arrow::Int64Type>;
+    }
+    case arrow::Type::FLOAT: {
+      return &ResolveTableFromScalar<arrow::FloatType>;
+    }
+    case arrow::Type::DOUBLE: {
+      return &ResolveTableFromScalar<arrow::DoubleType>;
+    }
+    case arrow::Type::HALF_FLOAT:break;
+    case arrow::Type::STRING:break;
+    case arrow::Type::BINARY:break;
+    case arrow::Type::FIXED_SIZE_BINARY:break;
+    case arrow::Type::DATE32:break;
+    case arrow::Type::DATE64:break;
+    case arrow::Type::TIMESTAMP:break;
+    case arrow::Type::TIME32:break;
+    case arrow::Type::TIME64:break;
+    case arrow::Type::INTERVAL:break;
+    case arrow::Type::DECIMAL:break;
+    case arrow::Type::LIST:break;
+    case arrow::Type::STRUCT:break;
+    case arrow::Type::UNION:break;
+    case arrow::Type::DICTIONARY:break;
+    case arrow::Type::MAP:break;
+    case arrow::Type::EXTENSION:break;
+    case arrow::Type::FIXED_SIZE_LIST:break;
+    case arrow::Type::DURATION:break;
+    case arrow::Type::LARGE_STRING:break;
+    case arrow::Type::LARGE_BINARY:break;
+    case arrow::Type::LARGE_LIST:break;
+  }
+
+  return nullptr;
+}
+
+cylon::Status Sum(const std::shared_ptr<cylon::Table> &table,
+                  int32_t col_idx,
+                  std::shared_ptr<cylon::Table> &output) {
+
+  cylon::Status status;
+  std::shared_ptr<cylon::compute::Result> result;
+  std::shared_ptr<cylon::CylonContext> ctx = table->GetContext();
+
+  if ((status = cylon::compute::Sum(table, col_idx, &result)).is_ok()) {
+    ResolveTableFromScalarFptr fptr = CreateTableFromScalar(result);
+    arrow::MemoryPool *memory_pool = cylon::ToArrowPool(ctx);
+    fptr(memory_pool, ctx, col_idx, result, table, output);
+  }
+  return status;
+}
+
+cylon::Status Count(const std::shared_ptr<cylon::Table> &table,
+                             int32_t col_idx,
+                             std::shared_ptr<cylon::Table> &output) {
+  cylon::Status status;
+  std::shared_ptr<cylon::compute::Result> result;
+  std::shared_ptr<cylon::CylonContext> ctx = table->GetContext();
+
+  if ((status = cylon::compute::Count(table, col_idx, &result)).is_ok()) {
+    ResolveTableFromScalarFptr fptr = CreateTableFromScalar(result);
+    arrow::MemoryPool *memory_pool = cylon::ToArrowPool(ctx);
+    fptr(memory_pool, ctx, col_idx, result, table, output);
+  }
+  return status;
+}
+
+cylon::Status Min(const std::shared_ptr<cylon::Table> &table,
+                           int32_t col_idx,
+                           std::shared_ptr<cylon::Table> &output) {
+  cylon::Status status;
+  std::shared_ptr<cylon::compute::Result> result;
+  std::shared_ptr<cylon::CylonContext> ctx = table->GetContext();
+
+  if ((status = cylon::compute::Min(table, col_idx, &result)).is_ok()) {
+    ResolveTableFromScalarFptr fptr = CreateTableFromScalar(result);
```

Comment:
check the above comment. Then, lines 419-421 would be a single function call. there's no need for a function pointer IMO.

Reply:
BTW we'll have to add another status check actually, to check if table creation was successful or not